pkg/asset/cluster/tfvars.go

```diff
@@ -325,6 +325,24 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 		}
 
 		preexistingnetwork := installConfig.Config.Azure.VirtualNetwork != ""
+
+		var bootstrapIgnStub, storageAccountName string
+		if installConfig.Azure.CloudName == azure.StackCloud {
+			// Our Azure Terraform config uses a random string to name storage accounts as the names must be globally unique & only alphanumeric.
+			// In order to create the bootstrap stub in advance, that name needs to be deterministic, so we can use the
+			// infrastructure ID suffix, which is also a random string. Keep the "cluster" prefix to be consistent with public Azure.
+			infraIDSuffix := strings.Split(clusterID.InfraID, "-")[1]
+			storageAccountName = fmt.Sprintf("cluster%s", infraIDSuffix)
```

Comment:
I am not sure that I am understanding the need for this. As far as I can tell, we don't need the storage account name to be deterministic. The only thing that we need to be deterministic is the string that we are searching for in the ignition stub to replace with the actual URL that we want to use for the bootstrap ignition config. The string that we search for does not have to be a URL: It can be a constant value unrelated to the specific cluster being installed.

Reply:
I think you are correct re constant string and that will be a great improvement. I will put it in.